### PR TITLE
[MM-28257] Cypress/E2E: Automate backlogs - Integrations > Slash Commands (10 test cases)

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
I did modify `loginAndVisitChannel` to support options in case some other test case want to use it too.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28257

### CODE REVIEW

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 1]
```diff
@@ -75,6 +77,127 @@ describe('Integrations', () => {
...
+            onBeforeLoad: (win) => {
+                cy.stub(win, 'open');
+            },
+        });
+
+        // # Type "/help"
+        cy.postMessage('/help');
+
+        // * Verify that a new tag opens
+        cy.window().its('open').should('be.called');
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:124]: I'd suggest to expand verification here like
```suggestion
        cy.window().its('open').should('have.been.calledWithMatch', 'https://about.mattermost.com/default-help/');
```

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 2]
```diff
@@ -75,6 +77,127 @@ describe('Integrations', () => {
...
+        cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Town Square');
+    });
+
+    it('MM-T683 /join', () => {
+        loginAndVisitChannel(user1, testChannelUrl);
+
+        // # Create New Channel
+        cy.apiCreateChannel(testTeam.id, 'new-channel', 'New Channel').then(({channel}) => {
+            // # Type "/join ~new-channel"
+            cy.postMessage(`/join ~${channel.name}`);
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:158]: This channel will be created by the `user1` itself and doing `/join` will not effectively do the command. I suggests below for other user to create a channel, and then be joined by `user1`
```suggestion
       // # Login as another user and create a new channel
        cy.apiAdminLogin();
        cy.apiCreateChannel(testTeam.id, 'new-channel', 'New Channel').then(({channel}) => {
            // # Login as test user and visit test channel
            loginAndVisitChannel(user1, testChannelUrl);

            // # Type "/join ~new-channel"
            cy.postMessage(`/join ~${channel.name}`);
```

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 3]
```diff
@@ -75,6 +77,127 @@ describe('Integrations', () => {
...
+    it('MM-T684 /me', () => {
+        loginAndVisitChannel(user1, testChannelUrl);
+
+        // # Type "/me test"
+        cy.postMessage('/me test');
+
+        // * Verify a message "test" is posted
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#post_${postId}`).find('.user-popover').should('have.text', user1.username);
+            cy.get(`#postMessageText_${postId}`).should('have.text', 'test');
```
Comment (saturninoabril) [inline e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:174]: Please add:
```suggestion
            cy.get(`#postMessageText_${postId}`).should('have.text', 'test');

            // * The message should match other system message formatting.
            cy.get(`#post_${postId}`).should('have.class', 'post--system');
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @hanzei! Please see few suggestions.

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js (hunk 1 above)
Comment (hanzei) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:124]: Good suggestion :+1:

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js [hunk 4]
```diff
@@ -75,6 +77,127 @@ describe('Integrations', () => {
...
+        loginAndVisitChannel(user1, testChannelUrl);
+
+        // # Use "/code"
+        cy.postMessage('/code 1. Not a list item, **not bolded**, http://notalink.com, ~off-topic is not a link to the channel.');
+
+        // * Verify that that markdown isn't rendered
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#post_${postId}`).find('.user-popover').should('have.text', user1.username);
+            cy.get(`#postMessageText_${postId}`).get('code').should('have.text', '1. Not a list item, **not bolded**, http://notalink.com, ~off-topic is not a link to the channel.');
+        });
```
[the remark this answers is not in the document]
Comment (hanzei) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:90]: Sure, I will add a test for that

**Comment by hanzei** [comment]
Thanks for the review @saturninoabril. PTAL

File: e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/integration/integrations/builtin_commands/common_commands_spec.js:90]: Thanks!

**Review by saturninoabril — APPROVED** [review]
Thanks Ben! LGTM, tested and passed!

**Comment by hanzei** [comment]
Swapping in  @srkgupta as a second reviewers to even out the number of open reviews. I hope you are fine with this @DHaussermann

**Review by srkgupta — CHANGES_REQUESTED** [review]
Hi @hanzei 

The tests looks great to me and are all working fine. Just one small change requested in both the testcase files:

Can you remove the `@prod` tag from both the testcase files as you have added new testcases to the existing file. These tests will be executed for 5 days and if it consistently passes, then QA team will promote them to `@prod`.

**Comment by hanzei** [comment]
@srkgupta Sure, done.

**Review by srkgupta — APPROVED** [review]
Thanks @hanzei. Approving the PR

**Pull request merged**